cmd/oras/internal/option/notty.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package option
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"golang.org/x/term"
+)
+
+const NoTTYFlag = "no-tty"
+
+// NoTTY option struct.
+type NoTTY struct {
+	noTTY bool
+	TTY   *os.File
+}
+
+// ApplyFlags applies flags to a command flag set.
+func (opts *NoTTY) ApplyFlags(fs *pflag.FlagSet) {
+	fs.BoolVarP(&opts.noTTY, NoTTYFlag, "", false, "[Preview] disable progress bars")
+}
+
+// Parse parses the input notty flag.
+func (opts *NoTTY) Parse(*cobra.Command) error {
+	// use STDERR as TTY output since STDOUT is reserved for pipeable output
+	if !opts.noTTY {
+		f := os.Stderr
+		if term.IsTerminal(int(f.Fd())) {
+			opts.TTY = f
+		}
+	}
+	return nil
+}
+
+// UpdateTTY updates the TTY value, given the status of --debug flag, --no-tty flag and output
+// path value.TTY value is set to nil if
+// 1. --no-tty flag is set to true
+// 2. --debug flag is used
+// 3. output path is set to stdout and --no-tty flag is not explicitly set to false
+// (i.e. not --no-tty==false)
+func (opts *NoTTY) UpdateTTY(debugEnabled, nottyFlagPresent, toSTDOUT bool) {
+	ttyEnforced := nottyFlagPresent && !opts.noTTY
+	if debugEnabled || opts.noTTY || (toSTDOUT && !ttyEnforced) {
+		opts.TTY = nil
+	}
+}
```

Comment:
I think this existing rule 3 is over complicated. We can simply it to "no tty if output == stdout". With the current version, if we run `oras blob fetch` (which is the only place it's used) with `--output - ` and explicitly `--notty=false`, the output is incorrectly displayed anyway.